l1-contracts/test/merkle/Merkle.t.sol

```diff
@@ -27,4 +51,125 @@ contract MerkleTest is Test {
       assertEq(merkle.computeRoot(), frontier.root(), "Frontier Roots should be equal");
     }
   }
+
+  function testNaiveSiblingPathAndMembershipVerification(uint256 _idx) public view {
+    uint256 leafIndex = bound(_idx, 0, merkle.SIZE() - 1);
+
+    (bytes32[] memory path, bytes32 leaf) = testNaiveMerkle.computeSiblingPath(leafIndex);
+
+    Merkle.verifyMembership(path, leaf, leafIndex, testNaiveMerkle.computeRoot());
+  }
+
+  function testNaiveBadSiblingPathAndMembershipVerification(uint256 _idx) public {
+    uint256 leafIndex = bound(_idx, 0, merkle.SIZE() - 1);
+    bytes32 expectedRoot = testNaiveMerkle.computeRoot();
+
+    // Tests garbled path
+    (bytes32[] memory path1, bytes32 leaf) = testNaiveMerkle.computeSiblingPath(leafIndex);
+    bytes32 temp1 = path1[0];
+    path1[0] = path1[path1.length - 1];
+    path1[path1.length - 1] = temp1;
+    vm.expectRevert();
+    this.verifyMembershipWrapper(path1, leaf, leafIndex, expectedRoot);
+
+    // Tests truncated path
+    (bytes32[] memory path2,) = testNaiveMerkle.computeSiblingPath(leafIndex);
+    bytes32[] memory truncatedPath = new bytes32[](path2.length - 1);
+    for (uint256 i = 0; i < truncatedPath.length; i++) {
+      truncatedPath[i] = path2[i];
+    }
+
+    vm.expectRevert();
+    this.verifyMembershipWrapper(truncatedPath, leaf, leafIndex, expectedRoot);
+
+    // Tests empty path
```

Comment:
I don't think having these 3 cases of empty, truncated and garbled path here really makes sense given the nature of merkle trees. Instead I would just use fuzzing and make the fuzzer provide the path on the input.

Reply:
All the fuzzer tests generate passing cases though, because we manually compute the correct path. I think it's important to have _some_ failure cases.